prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -4146,16 +4140,16 @@ subroutine readTemperatureProfile(node, modifier, ctrl)
     call destruct(li1)
     call destruct(lr1)
     allocate(ctrl%tempMethods(size(tmpC1)))
-    lp2: do ii = 1, size(tmpC1)
-      do jj = 1, size(tempMethodNames)
-        if (trim(tmpC1(ii)) == tolower(trim(tempMethodNames(jj)))) then
-          ctrl%tempMethods(ii) = jj
-          cycle lp2
+    do ii = 1, size(tmpC1)
+      block
+        logical :: success
```

Comment:
Is this really worth introducing a `block` here? We already refactored the parser to have local scopes by subroutines and since we are in `readTemperatureProfile` I don't see a need for a separate `block` here.